lib/hanami/routes.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Hanami
+  # Hanami application routes
+  #
+  # @since 2.0.0
+  class Routes
+    attr_reader :apps
+
+    def initialize(&blk)
+      @blk = blk
+      @apps = []
+      instance_eval(&blk)
+    end
+
+    def to_proc
+      @blk
+    end
+
+    def mount(app, **)
+      @apps << app if app.is_a?(Symbol)
```

Comment:
Feels like this should fail more noisily if it's ever passed something besides a symbol?

Reply:
@timriley Please consider that this is **NOT** exposed directly to users. This is a trick to extract the Hanami _slices_ mounted inside `config/router.rb`.

However:

  1. The object tha powers directly `config/routes.rb` is `Hanami::Router#mount` redefined by `lib/hanami/frameworks.rb` in this PR.
  2. That `#mount` API must allow non-symbol arguments to mount Rack apps. (e.g. `mount MySinatraApp.new`)

Reply:
@jodosha OK, with this extra info it makes more sense to me how this particular method works, thanks.

So this particular Hanami::Routes class has 2 key jobs: (a) holds the block that is passed when it is initialized (so it can be passed to the real router later on?) and (b) holds a set of apps/slices, so the app can know what has been mounted? Is this right?

Reply:
What's still not clear is why we need this `method_missing(*)` implementation?